pretty_help/pretty_help.py

```diff
@@ -329,9 +336,26 @@ async def send_bot_help(self, mapping):
         ctx = self.context
         bot = ctx.bot
 
-        if bot.description:
-            # <description> portion
-            self.paginator.add_line(self._no_category, bot.description, empty=True)
+        if self.show_index:
+            if bot.description:
+                self.paginator.add_line(self._index, bot.description, empty=True)
+
+            get_width = discord.utils._string_width
+            cogs = bot.cogs
+            max_size = max(map(len, cogs))
+            for cog_name, cog in cogs.items():
+                width = max_size - (get_width(cog_name) - len(cog_name))
+                description = cog.description if cog.description else ''
+                entry = "{0}{1:<{width}} {2}".format(
+                    self.indent * " ", cog_name, description, width=width
+                )
+                self.paginator.add_line(self._index, self.shorten_text(entry))
+
+            self.paginator.add_line(self._no_category, '', empty=True)
+        else:
+            if bot.description:
+                # <description> portion
+                self.paginator.add_line(self._no_category, bot.description, empty=True)
```

Comment:
I like it. Just a few things:
* Revise so the output is alphabetical
* Use the `qualified_name` of the cog just to be safe.
* Check if there are even any cogs loaded in the conditional

``` python
        if self.show_index and bot.cogs:
            if bot.description:
                self.paginator.add_line(self._index, bot.description, empty=True)

            get_width = discord.utils._string_width
            cogs = bot.cogs
            max_size = max(map(len, cogs))
            for cog in sorted(cogs.values(), key=lambda c: c.qualified_name):
                cog_name = cog.qualified_name
                width = max_size - (get_width(cog_name) - len(cog_name))
                description = cog.description if cog.description else ""
                entry = "{0}{1:<{width}}| {2}".format(
                    self.indent * " ", cog_name, description, width=width
                )
                self.paginator.add_line(self._index, self.shorten_text(entry))

        else:
            if bot.description:
                # <description> portion
                self.paginator.add_line(self._no_category, bot.description, empty=True)
```

Reply:
Thank you!
I see that you removed the line
`            self.paginator.add_line(self._no_category, '', empty=True)`
Which I had added to ensure that uncategorized commands show up right after the index instead of in a random place.
Also, how about sorting the cogs based on the value of `self.sort_commands`?

`        if self.show_index and bot.cogs:
            if bot.description:
                self.paginator.add_line(self._index, bot.description, empty=True)

            get_width = discord.utils._string_width
            cogs = bot.cogs
            max_size = max(map(len, cogs))
            cog_objects = sorted(cogs.values(), key=lambda c: c.qualified_name) if self.sort_commands else cogs.values()
            for cog in cog_objects:
                cog_name = cog.qualified_name
                width = max_size - (get_width(cog_name) - len(cog_name))
                description = cog.description if cog.description else ""
                entry = "{0}{1:<{width}}| {2}".format(
                    self.indent * " ", cog_name, description, width=width
                )
                self.paginator.add_line(self._index, self.shorten_text(entry))

            self.paginator.add_line(self._no_category, '', empty=True)
        else:
            if bot.description:
                # <description> portion
                self.paginator.add_line(self._no_category, bot.description, empty=True)`

Reply:
Good call, let's do that.

I removed the line ` self.paginator.add_line(self._no_category, '', empty=True)` because I'm working on a conditional *No Category* and that would create the page every time. Just trying to figure out how to handle multiple changes lol

Reply:
Alright, I'll let you handle that line. I've pushed the changes.